src/main/java/core/basesyntax/validator/quantity/QuantityValidatorImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.validator.quantity;
+
+public class QuantityValidatorImpl implements QuantityValidator {
+    @Override
+    public boolean isQuantityCorrectForPurcase(long quantity,
+                                               long fruitBalace, int lineNumber) {
+        try {
+            if (quantity > fruitBalace) {
+                throw new UnavailableQuantity();
+            }
+            return true;
+        } catch (UnavailableQuantity e) {
```

Comment:
this doesn't make sense

why are you throwing exception and then catching it?
let's just throw it - and that's all
extend your custom exception from RuntimeException